export/mlab_export.py

```diff
@@ -0,0 +1,613 @@
+    'must equal the value in the collectd config. Inaccurate values will not '
+    'change the intervals of exported data, but could result in samples being '
+    'skipped at the beginning or end of the export window.', lower_bound=1)
+flags.DEFINE_integer(
+    'ts_start', None,
+    'Timestamp to start export, in seconds since the epoch. Only use this '
+    'option for debugging. Normally ts_start is calculated automatically from '
+    'the previous export end time. If given, ts_start should be a multiple of '
+    'step', lower_bound=0)
+flags.DEFINE_integer(
+    'ts_end', None,
+    'Timestamp to end export, in seconds since the epoch. Only use this option '
+    'for debugging. Normally, ts_end is calculated automatically from: '
+    'ts_start + length.', lower_bound=0)
+flags.DEFINE_integer(
+    'ts_offset', 600, 'Amount of time (seconds) that must have passed after '
+    'ts_end to ensure that values cached by collectd have been flushed to disk '
+    'before attempting an export.', lower_bound=0)
+flags.DEFINE_multistring(
+    'ignored_experiments', [], 'List of experiment names to ignore. Experiment '
+    'must be in "slice.site" form not "site_slice".')
+flags.DEFINE_bool(
+    'pretty_json', None, 'Add extra indenting to json output (for debugging).')
+flags.DEFINE_string(
+    'output_dir', EXPORT_DIR, 'Root directory of json output files.')
+flags.DEFINE_string(
+    'output', None, 'Name of json output file. Set automatically if not given.')
+flags.DEFINE_string(
+    'export_metrics', METRIC_MAP_CONF, 'File name with metric map. The metric '
+    'map defines canonical metric names for raw, metric names taken from '
+    'collectd RRD files.')
+flags.DEFINE_bool(
+    'verbose', False, 'Increase verbosity level.')
+flags.DEFINE_bool(
+    'show_nagios', False, 'Shows collectd-nagios commands to monitor metrics.')
+flags.DEFINE_bool(
+    'show_rrdfile', False, 'Shows the RRD files opened during export.')
+flags.DEFINE_bool(
+    'show_metric', False, 'Shows the canonical metric names during export.')
+flags.DEFINE_bool(
+    'show_rawmetric', False, 'Shows the raw metric name before translating to '
+    'the canonical name. This can be helpful to know before adding new metrics '
+    'to the export metrics configuration file.')
+flags.DEFINE_bool(
+    'show_skipped', False, 'Shows the raw metric names that are not exported. '
+    'This option may be helpful when adding new metrics to the export_metrics.')
+flags.DEFINE_bool(
+    'update', True, 'Update timestamps on successful export. Update is always '
+    'disabled when any --show_* option is enabled.')
+flags.DEFINE_bool(
+    'compress', False,
+    'Compresses output and adds .gz extension to output filename.')
+
+
+class Error(Exception):
+  """Base error type for this file."""
+
+
+class TimeOptionError(Error):
+  """An error related to export times or ranges."""
+
+
+class LockFileError(Error):
+  """An error related to the single process lock file."""
```

Comment:
Optional, fairly subjective: Empty classes look weird to me if they don't contain `pass`. The style guide doesn't seem to explicitly require this, though they do use a `pass` in their example:

https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Exceptions#Exceptions

This might just be because they needed some statement and your triple string makes the `pass` unnecessary.